rotkehlchen/exchanges/coinbase.py

```diff
@@ -1042,8 +1048,23 @@ def _deserialize_history_event(self, raw_data: dict[str, Any]) -> HistoryEvent |
             asset = asset_from_coinbase(amount_data['currency'], time=timestamp)
             notes = raw_data.get('details', {}).get('header', '')
             tx_type = raw_data['type']
+            if tx_type in ('staking_transfer', 'unstaking_transfer'):
+                event_subtype = HistoryEventSubType.DEPOSIT_ASSET if tx_type == 'staking_transfer' else HistoryEventSubType.REMOVE_ASSET  # noqa: E501
+                return HistoryEvent(
+                    event_identifier=f'{CB_EVENTS_PREFIX}{raw_data["id"]!s}',
+                    sequence_index=0,
+                    timestamp=ts_sec_to_ms(timestamp),
+                    location=Location.COINBASE,
+                    event_type=HistoryEventType.STAKING,
+                    event_subtype=event_subtype,
+                    asset=asset,
+                    balance=Balance(amount),
+                    location_label=None,
+                    notes=f'{notes}as {tx_type.replace("_", " ")}' if notes != '' else notes,
+                )
+
             if notes != '':
-                notes += f' {"from coinbase earn" if tx_type == "send" else "as " + tx_type}'
+                notes += ' from coinbase card cashback' if tx_type == 'cardbuyback' else f' as {tx_type}'  # noqa: E501
```

Comment:
why change this? What does earn and card cashback have to do with each other? These are different programs.